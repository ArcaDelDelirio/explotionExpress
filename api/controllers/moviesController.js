"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const express_1 = require("express");
const router = express_1.Router();
exports.moviesController = router;
//# sourceMappingURL=moviesController.js.map