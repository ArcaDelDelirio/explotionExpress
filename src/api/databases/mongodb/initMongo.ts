import { connect } from 'mongoose';

export class SingleMongo {
    private readonly dbURI: string;
    public constructor(url: string, collection: string) {
        this.dbURI = `mongodb://${url}/${collection}`;
    }

    public connectToMongo() {
        return connect(this.dbURI);
    }
}
