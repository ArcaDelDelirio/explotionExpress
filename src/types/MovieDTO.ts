export default interface MovieDTO {
    title: string;
    year: number;
    genre: string[];
    duration: number;
}
