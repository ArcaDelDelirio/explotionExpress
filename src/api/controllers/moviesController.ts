import { MovieRepository } from '../repo/movieRepository';
import { MovieDTO } from 'MovieDTO';

export class MoviesController {
    private repo: MovieRepository;
    /**
     * Init Repository for Movies
     * @returns MovieRepository
     */
    public initDatabase() {
        this.repo = new MovieRepository();
        return this.repo;
    }

    private static parseInput(data: any): MovieDTO {
        return {
            title: data.title,
            year: data.year,
            genre: data.genre,
            duration: data.duration,
        };
    }

    public async showMovies() {
        return await this.repo.showMovies();
    }
}
