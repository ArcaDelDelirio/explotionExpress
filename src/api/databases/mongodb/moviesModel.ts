import { Document, Model, model, Schema } from 'mongoose';
import { MovieDTO } from 'MovieDTO';

export interface MovieModel extends MovieDTO, Document {
    getTitleAndYear(): string;
}
export const MovieSchema: Schema = new Schema({
    title: String,
    year: Number,
    genre: [String],
    duration: Number,
});

MovieSchema.methods.getTitleAndYear = function(): string {
    return this.title + ' ' + this.duration;
};

export const Movie: Model<MovieModel> = model<MovieModel>('movies', MovieSchema);
