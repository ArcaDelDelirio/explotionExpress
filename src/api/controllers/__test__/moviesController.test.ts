import { MoviesController } from '../moviesController';
import { MovieDTO } from 'MovieDTO';
import { SingleMongo } from '../../databases/mongodb/initMongo';
import { Connection, Mongoose } from 'mongoose';

describe('MoviesController', () => {
    const data: MovieDTO = {
        title: 'Glass',
        year: 2019.0,
        genre: ['Terror', 'Suspense'],
        duration: 129.0,
    };
    describe('createMovie', () => {
        test('should return ID of save', () => {});
    });
    describe('showMovies', () => {
        test('should return json movies', () => {
            MoviesController.initDatabase()
            const movies = MoviesController.showMovies();
            expect(movies).toBe('Not implemented');
        });
    });
    describe('findMovieByName', () => {
        test('', () => {});
    });
    describe('findMovieByTitle', () => {
        test('', () => {});
    });
    describe('findMovieById', () => {
        test('', () => {});
    });
    describe('findMoviesByYear', () => {
        test('', () => {});
    });
    describe('findMoviesByGenre', () => {
        test('', () => {});
    });
    describe('findMoviesByDuration', () => {
        test('', () => {});
    });
});
