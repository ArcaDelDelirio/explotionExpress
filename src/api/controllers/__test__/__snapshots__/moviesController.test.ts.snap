// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesController Parse Data to Json Validate Data for Database 1`] = `
Array [
  Object {
    "_id": "5d246d902329923d43151365",
    "duration": 100,
    "genre": Array [
      "Comedy",
      "Animation",
    ],
    "title": "Toy Story 4",
    "year": 2019,
  },
  Object {
    "_id": "5d246d902329923d43151366",
    "duration": 181,
    "genre": Array [
      "SuperHero",
      "Action",
    ],
    "title": "Vengadores:Endgame",
    "year": 2019,
  },
  Object {
    "_id": "5d246d902329923d43151367",
    "duration": 129,
    "genre": Array [
      "Terror",
      "Suspense",
    ],
    "title": "Glass",
    "year": 2019,
  },
]
`;
